### ISSUE TITLE
Make "history" return by fit store the initial losses

### ISSUE THREAD

**Issue by makecent** [tensorflow/tensorflow#38591]
**System information**
- TensorFlow version (you are using): 2.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

Currently, the returned `history `class by `fit` store the losses start from loss after first epoch training. Hence it doesn't contain the initial loss before training start, which is very important for investigating the influence of first epoch training and the impact of different initialization methods. 

Besides, in this case, the epoch indexes stored in `history` are not aligned with the real training epoch because epoch 0 in `history` corresponds to epoch 1 in real training. If we set epoch 0 in history to represent the initial state, things will look more reasonable. 

Therefore I suggest making the `history` class return by `fit` stores the initial losses, on both train and validation set,  and flag it as epoch 0 by default.

**Will this change the current api? How?** 

It depends. Either add an extra argument to the fit function or make this as default so no need to change the current usage of API.

**Who will benefit with this feature?**
All users want to log the initial loss. And those who want to do something with 'history'

**Comment by ravikyram** [comment]
@MakeCent 

Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by MakeCent** [comment]
@ravikyram ANYBODY can benifit from this acturally. It make the epoch indexes rensonable and give a complete view of training process. I will try to write codes to see if I can handle this and then, if done, I will submist a PR. Thanks!

**Comment by jvishnuvardhan** [comment]
@makecent Sorry for the late response. I understand it may help investigating the effect of different initializations. Other than that, which use-cases will need this kind of feature? Also, i think you can write a custom keras callback to get the initial metrics.

Please note that Keras development moved to [keras-team/keras repo.](https://github.com/keras-team/keras/issues repository to focus on only keras. If you want to open any PR, please feel free to open in that repository. Thanks!
